tests/flytekit/unit/annotated/test_serialization.py

```diff
@@ -97,6 +97,42 @@ def my_wf(a: int, b: str) -> (int, str):
         assert wf.nodes[2].branch_node is not None
 
 
+def test_serialization_branch_complex_2():
+    @task
+    def t1(a: int) -> typing.NamedTuple("OutputsBC", t1_int_output=int, c=str):
+        return a + 2, "world"
+
+    @task
+    def t2(a: str) -> str:
+        return a
+
+    @workflow
+    def my_wf(a: int, b: str) -> (int, str):
+        x, y = t1(a=a)
+        d = (
+            conditional("test1")
+            .if_(x == 4)
```

Comment:
Can we add two more test cases?  one where the if_ statement has an `||` in it.  and also, one where the then statements call subworkflows?